pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -186,6 +205,28 @@ func (r *ReconcileEgressGateway) Reconcile(ctx context.Context, request reconcil
 			reqLogger.Info("EgressGateway object not found")
 			// Since the EGW resource is not found, remove the deployment.
 			r.status.RemoveDeployments(types.NamespacedName{Name: request.Name, Namespace: request.Namespace})
+
+			// In case of openShift, we are using a single SCC.
+			// Whenever a EGW resource is deleted, remove the corresponding user from the SCC
+			// and update the resource.
+			if r.provider == operatorv1.ProviderOpenShift {
```

Comment:
We really should generate the list of names we need to create and pass that to the `render` package as part of the Config struct - that would better follow our established paradigms. 

e.g.,

```go
	config := &egressgateway.Config{
		PullSecrets:  pullSecrets,
		Installation: installation,
		OSType:       rmeta.OSTypeLinux,
		EgressGW:     egw,
		VXLANPort:    egwVXLANPort,
		VXLANVNI:     egwVXLANVNI,
		Openshift:    openshift,
		UsePSP:       r.usePSP,
                EgressGWNames: getEGWNames(egws),
	}

func getEGWNames(egws []v1.EgressGateway) []string { ... }
```

The render logic would then build the SCC, sort the given names (so that the output of the render package is deterministic), and include them in the produced SCC as serviceaccounts.

Reply:
Does that mean, I don't need to create this as a Passthrough object? I think, passing egwNames to the SecurityContextConstraints directly will be beneficial. The reason being, when a EGW resource gets deleted, SCC must be updated and having egwNames outside the config can be helpful right? WDYT?